### PR TITLE
[ADD] Construction: added construction industry

### PR DESCRIPTION
**Pull request by arpa-odoo**
added construction industry

task-3686809

### CODE REVIEW

File: construction/static/src/binary/hr_employee/1-image_1920
Comment (vava-odoo) [inline construction/static/src/binary/hr_employee/1-image_1920:1]: In hr_employee.xml only images starting with 4, 5 and 7 are used. You can maybe remove others?

File: construction/static/src/binary/res_partner/18-image_1920
Comment (vava-odoo) [inline construction/static/src/binary/res_partner/18-image_1920:1]: Not used?

File: construction/static/src/binary/res_partner/14-image_1920
Comment (vava-odoo) [inline construction/static/src/binary/res_partner/14-image_1920:1]: Not used?

**Review by vava-odoo — CHANGES_REQUESTED** [review]
Hello @arpa-odoo 
Thanks for this, it works as expected :+1: 
Could you have a look at the following comments? There is a lot of them, but mostly minor ones.
Cheers!

File: construction/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        'helpdesk_fsm',
+        'helpdesk_sale',
+        'hr_fleet',
+        'hr_holidays',    
+        'industry_fsm_sale_report',
+        'industry_fsm_stock',
+        'knowledge',
+        'maintenance',
+        'purchase_stock',
+        'stock_barcode',
```
Comment (vava-odoo) [inline construction/__manifest__.py:25]: No barcorde
```suggestion
```

File: construction/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        'appointment_account_payment',
+        'appointment_crm',
+        'crm_enterprise',
+        'documents_hr',
+        'documents_project_sale',
+        'documents_spreadsheet',
+        'helpdesk_fsm',
+        'helpdesk_sale',
+        'hr_fleet',
+        'hr_holidays',
```
Comment (vava-odoo) [inline construction/__manifest__.py:19]: no time off
```suggestion
```

File: construction/data/documents_folder.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="documents_folder_11" model="documents.folder">
+        <field name="name">Project Template</field>
+        <field name="parent_folder_id" ref="documents_project.documents_project_folder"/>
+    </record>
+    <record id="documents_folder_12" model="documents.folder">
+        <field name="name">S00001 - VDK Project</field>
+        <field name="parent_folder_id" ref="documents_project.documents_project_folder"/>
+    </record>
```
[the remark this answers is not in the document]
Comment (arpa-odoo) [reply construction/data/documents_folder.xml:10]: okay

File: construction/data/project_task_type.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        <field name="name">Done</field>
+        <field name="sequence">20</field>
+        <field name="fold" eval="True"/>
+    </record>
+    <record id="planning_project_stage_4" model="project.task.type">
+        <field name="name">Cancelled</field>
+        <field name="sequence">25</field>
+        <field name="fold" eval="True"/>
+    </record>
+</odoo>
```
[the remark this answers is not in the document]
Comment (arpa-odoo) [reply construction/data/project_task_type.xml:24]: yes you are correct but in the fsm_demo file this all records was unlinked..so we cant use it.

File: construction/demo/fleet_cars_data.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="fleet.model_classa" model="fleet.vehicle.model">
```
[the remark this answers is not in the document]
Comment (arpa-odoo) [reply construction/demo/fleet_cars_data.xml:3]: okay

File: construction/demo/project_update.xml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="project_update_1" model="project.update" context="{'default_project_id': ref('project_project_4')}">
```
[the remark this answers is not in the document]
Comment (arpa-odoo) [reply construction/demo/project_update.xml:3]: no by using only project_id i got a error
null value in column "project_id" of relation "project_update" violates not-null constraint

File: construction/data/project_task_type.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (vava-odoo) [reply construction/data/project_task_type.xml:24]: But fsm_demo is only in demo, so it won't affect the industry (standard modules are without demo)

File: construction/demo/project_update.xml (hunk 6 above)
[the remark this answers is not in the document]
Comment (vava-odoo) [reply construction/demo/project_update.xml:3]: ok

File: construction/demo/project_task.xml [hunk 7]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo noupdate="1">
+    <record id="project_task_10" model="project.task">
+        <field name="name">Down Payment</field>
+        <field name="user_ids" eval="[Command.set([ref('base.user_admin')])]"/>
+        <field name="project_id" ref="project_project_4"/>
+        <field name="stage_id" ref="industry_fsm.planning_project_stage_3"/>
+        <field name="allocated_hours">1.0</field>
+        <field name="sale_line_id" ref="sale_order_line_4"/>
+    </record>
```
Comment (vava-odoo) [inline construction/demo/project_task.xml:10]: A task down payment?! It looks like a bug (in standard) 
```suggestion
```

File: construction/demo/res_partner.xml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    <record id="res_partner_13" model="res.partner">
+        <field name="name">Wood Corner</field>
+        <field name="is_company" eval="True"/>
+        <field name="image_1920" type="base64" file="construction/static/src/binary/res_partner/13-image_1920"/>
+    </record>
+    <record id="base.main_partner" model="res.partner">
+        <field name="name">My Construction Business</field>
+        <field name="is_company" eval="True"/>
+        <field name="country_id" ref="base.be"/>
+        <field name="email">cth@example.com</field>
```
Comment (vava-odoo) [inline construction/demo/res_partner.xml:43]: Maybe best to change that, he wants to remain incognito :smile: 
```suggestion
        <field name="email">ceth@example.com</field>
```

File: construction/demo/account_analytic_account.xml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
```
Comment (vava-odoo) [inline construction/demo/account_analytic_account.xml:2]: Please bring back "Internal" one + related project.task and account.analytic.line

File: construction/static/src/binary/product_template/3-image_1920
Comment (vava-odoo) [inline construction/static/src/binary/product_template/3-image_1920:1]: Don't see where it is used

File: construction/static/src/binary/product_template/4-image_1920
Comment (vava-odoo) [inline construction/static/src/binary/product_template/4-image_1920:1]: Don't see where it is used

**Review by vava-odoo — CHANGES_REQUESTED** [review]
Thanks for the update. It looks good and we should be merging really soon.
Please avoid losing data between reviews. Really difficult to track...
Cheers!

File: construction/demo/documents_document.xml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="document_project_project_4" model="documents.document">
+        <field name="name">machine.png</field>
+        <field name="datas" type="base64" file="construction/static/src/binary/ir_attachment/machine.jpg"/>
+        <field name="folder_id" ref="documents_folder_12"/>
+        <field name="res_model">project.project</field>
+        <field name="res_id" ref="project_project_4"/>
+        <field name="owner_id" ref="base.user_admin"/>
+    </record>
```
Comment (vava-odoo) [inline construction/demo/documents_document.xml:10]: Could you also remove weird standard data?
```xml
    <record id="documents.documents_image_multi_pdf_document" model="documents.document">
        <field name="active" eval="False"/>
    </record>
    <record id="documents.documents_mail_png" model="documents.document">
        <field name="active" eval="False"/>
    </record>
```

**Review by vava-odoo — APPROVED** [review]

**Pull request merged**